kasa/smartdevice.py

```diff
@@ -325,10 +329,16 @@ async def _modular_update(self, req: dict) -> None:
             )
             self.add_module("emeter", Emeter(self, self.emeter_type))
 
-        for module in self.modules.values():
+
+        for module_key in self.modules:
+            module = self.modules[module_key]
```

Comment:
```suggestion
        for module_name, module in self.modules.items():
```
Doing something like this avoids the extra assignment while still getting the key.